src/server/snapshot.h

```diff
@@ -144,6 +144,8 @@ class SliceSnapshot {
   std::atomic_bool closed_chan_{false};  // true if dest_->StartClosing was already called
 
   DbIndex current_db_;
+
+  // TODO : drop default_buffer from this class, we dont realy need it.
   std::unique_ptr<io::StringFile> default_buffer_;  // filled by default_serializer_
```

Comment:
That's true, because currently we always move out of the initializer

However, in an optimized version, we shouldn't do this and should keep the default buffer.

With it, we can make use of the following:
- If we do compression, then we already copy the compressed part into a new string, so there is no reason the steal the default buffer and then drop it
- In case of big entries, a fresh string file will grow like a vector in steps, increasing the capacity twofold. This amounts to unnecessary intermediate allocations and a buffer that can be twice as large as needed. Instead, we should copy the default buffers contents, and just reset it. Once its large enough, we will read all entries without intermediate alloactions

Reply:
Actually, instead of using FlushToSink at all, we should be able to get the io::Bytes/string_view from the serializer directly, so that we can do compression without an intermediate copy

But I guess that's another topic, off from the current discussion

Reply:
I am going to move the compression under RdbSerilaizer, so the flow will change, you can see the PR created already.
I agree on the the optimization for not allocating new buffer for each copy, I believe the flow will change more till I get to this TODO but in the current flow I believe we can call it tmp_buffer_ and use it in the temporary Serializer as will, right?
Anyway I am not going to change this now